spec/std/set_spec.cr

```diff
@@ -421,4 +421,97 @@ describe "Set" do
       set.clone.compare_by_identity?.should be_true
     end
   end
+
+  describe "keep_if" do
+    it "keeps only elements that evaluate to true" do
+      set = Set{1, 2, 3}
+      set.keep_if { |n| n < 2 }
+      set.should eq(Set{1})
+    end
+
+    it "returns self" do
+      set = Set{1, 2, 3}
+      set.keep_if { |n| n < 2 }.should be(set)
+    end
+  end
+
+  describe "select!" do
+    it "keeps only elements that evaluate to true" do
+      set = Set{1, 2, 3}
+      set.select! { |n| n < 2 }
+      set.should eq(Set{1})
+    end
+
+    it "returns self if changes were made" do
+      set = Set{1, 2, 3}
+      set.select! { |n| n < 2 }.should eq(set)
```

Comment:
This spec could just as well return a copy and the test would still pass. Do we need to also check that the underlying hash is actually the same?

(the underlying hash, as `Set` is a struct and is thus copied around)

Reply:
That's the whole point of #12289 and #12297 (some overloads of `#same?` are still missing)